File: core/amber/src/main/scala/edu/uci/ics/texera/web/resource/auth/PasswordEncryption.java
package edu.uci.ics.texera.web.resource.auth;

import java.io.UnsupportedEncodingException;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

import sun.misc.BASE64Encoder;

public class PasswordEncryption {
    public static String EncoderByMd5(String str) throws NoSuchAlgorithmException, UnsupportedEncodingException{
        // the way to encrypt the password
        MessageDigest md5=MessageDigest.getInstance("MD5");
        BASE64Encoder base64en = new BASE64Encoder();

        String newstr=base64en.encode(md5.digest(str.getBytes("utf-8")));
        return newstr;
    }

}

File: core/amber/src/main/scala/edu/uci/ics/texera/web/auth/PasswordEncryption.scala
package edu.uci.ics.texera.web.auth

import org.jasypt.util.password.StrongPasswordEncryptor

object PasswordEncryption {
  def encrypt(str: String): String = {
    val passwordEncryptor = new StrongPasswordEncryptor
    passwordEncryptor.encryptPassword(str)
  }
  def checkPassword(encryptedPassword: String, inputPassword: String): Boolean = {
    val passwordEncryptor = new StrongPasswordEncryptor
    passwordEncryptor.checkPassword(inputPassword, encryptedPassword)
  }
}
